src/utils.js

```diff
@@ -3,16 +3,55 @@
 
 /**
  * @author lattmann / https://github.com/lattmann
+ * @author pmeijer / https://github.com/pmeijer
+ * @author brollb / https://github.com/brollb
  */
 
 'use strict';
 
 var fs = require('fs'),
     Q = require('q'),
-    genDecoratorSvgList = require('./client/assets/generate_decorator_svg_list'),
-    ncp = require('ncp'), // Module for copying entire directory
     path = require('path'),
-    requireUncached = require('require-uncached');
+    requireUncached = require('require-uncached'),
+    SVGMapDeffered;
+
+function walkDir (dir, done) {
+    var deferred = Q.defer(),
+        results = [];
+
+    fs.readdir(dir, function (err, list) {
+        if (err) {
+            deferred.reject(err);
+            return;
+        }
+        var pending = list.length;
+        if (!pending) {
+            deferred.resolve(results);
+            return;
+        }
+
+        list.forEach(function (file) {
+            file = path.join(dir, file);
+            fs.stat(file, function (err, stat) {
+                if (stat && stat.isDirectory()) {
+                    walkDir(file, function (err, res) {
```

Comment:
promisified (and for clarity, probably the fs.stat could be called as a promise)